src/pivot-table/hooks/use-selections-model.ts

```diff
@@ -48,68 +86,68 @@ export default function useSelectionsModel(
   }, [selections, updatePageInfo]);
 
   const isLocked = useCallback(
-    (qType: EngineAPI.NxSelectionCellType, qRow: number, qCol: number) => {
-      switch (qType) {
-        case NxSelectionCellType.NX_CELL_LEFT:
-          return !!selected.find(
-            (cell) =>
-              cell.qType === NxSelectionCellType.NX_CELL_TOP ||
-              (cell.qType === NxSelectionCellType.NX_CELL_LEFT && cell.qCol !== qCol),
-          );
-        case NxSelectionCellType.NX_CELL_TOP:
-          return !!selected.find(
-            (cell) =>
-              cell.qType === NxSelectionCellType.NX_CELL_LEFT ||
-              (cell.qType === NxSelectionCellType.NX_CELL_TOP && cell.qRow !== qRow),
-          );
-        default:
-          return false;
+    (cell: Cell) => {
+      if (selectedPivotCells.size === 0 || selectedRowOrColumn.selectionCellType === null) {
+        return false;
       }
+
+      if (cell.selectionCellType !== selectedRowOrColumn.selectionCellType) {
+        return true;
+      }
+
+      if (cell.selectionCellType === NxSelectionCellType.NX_CELL_LEFT) {
+        return selectedRowOrColumn.coord !== cell.x;
+      }
+
+      return selectedRowOrColumn.coord !== cell.y;
```

Comment:
Maybe this is like a qlik specific question, but out of curiosity + i don't understand this part that why not matching coords of a cell would means that cell is locked or not? 

I see (kinda) same logic in the previous version of `isLocked` implementation as well

Reply:
You can only do selections in one dimension field at the time. So if there are ongoing selections in a field. All other fields are locked.

But since field information is not available here. We use the `selectionCellType` and `coord` to figure out which field have ongoing selections.

Reply:
can we have a comment about it, or it's to simple to have it like this (because i get confused tbh for a sec 😰)